site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md

```diff
@@ -0,0 +1,469 @@
+The [`"minimum_chrome_version"`][manifest-min-version] explains how this key behaves when a user tries to install your extension but isn't using a compatible version of Chrome.
+
+{% endAside %}
+
+The following code registers the omnibox event listeners at the top level of the script and updates [chrome.storage][api-storage] with the most recent api search.
+
+{% Label %}sw-omnibox.js:{% endLabel %}
+
+```js
+...
+const URL_CHROME_EXTENSIONS_DOC =
+  'https://developer.chrome.com/docs/extensions/reference/';
+const NUMBER_OF_PREVIOUS_SEARCHES = 4;
+
+// Display the suggestions after user starts typing
+chrome.omnibox.onInputChanged.addListener(async (input, suggest) => {
+  await chrome.omnibox.setDefaultSuggestion({
+    description: 'Enter a Chrome API or choose from past searches'
+  });
+  const { apiSuggestions } = await chrome.storage.local.get('apiSuggestions');
+  const suggestions = apiSuggestions.map((api) => {
+    return { content: api, description: `Open chrome.${api} API` };
+  });
+  suggest(suggestions);
+});
+
+// Open the reference page of the chosen API
+chrome.omnibox.onInputEntered.addListener((input) => {
+  chrome.tabs.create({ url: URL_CHROME_EXTENSIONS_DOC + input });
+  // Save the latest keyword
+  updateHistory(input);
+});
+
+async function updateHistory(input) {
+  const { apiSuggestions } = await chrome.storage.local.get('apiSuggestions');
+  apiSuggestions.unshift(input);
+  apiSuggestions.splice(NUMBER_OF_PREVIOUS_SEARCHES);
+  await chrome.storage.local.set({ apiSuggestions });
+}
+```
+
+{% Aside 'important' %}
+
+Extension service workers have access to both web APIs and Chrome APIs, with a few exceptions.
+For a deep dive, see [Service Workers...](tbd) 
+
+{% endAside %}
+
+### Step 6: Set up a recurring event {: #step-6 }
+
+The `setTimeout()` or `setInterval()` methods are commonly used to perform delayed or periodic
+tasks. However, these APIs can fail because the scheduler will cancel the timers when the service
+worker is terminated. Instead, extensions can use the [`chrome.alarms`][api-alarms] API. 
+
+To use the Alarms API, request the `"alarms"` permission in the manifest. The extension also needs to request [host permission][doc-host-perm] to retrieve the extension tips from the glitch site:
+
+{% Label %}manifest.json:{% endLabel %}
+
+```json/2/3
+{
+  ...
+  "permissions": ["storage", "alarms"],
+  "permissions": ["storage"],
+  "host_permissions": ["https://extension-tips.glitch.me/*"],
+}
+```
+
+The following code sets up an alarm once a day to fetch the daily tip and save it to
+[`chrome.storage.local()`][api-storage]:
+
+{% Label %}sw-tips.js:{% endLabel %}
+
+```js
+// Fetch tip & save in storage
+const updateTip = async () => {
+  const response = await fetch('https://extension-tips.glitch.me/tips.json');
+  const tips = await response.json();
+  const randomIndex = Math.floor(Math.random() * tips.length);
+  await chrome.storage.local.set({ tip: tips[randomIndex] });
```

Comment:
remove the await here, but return the promise from set

Reply:
I'm going to go through and update the code to match the sample repo. On it! 🏃🏻‍♀️💨